### PR TITLE
#157828773 Implementation of signup for users and middlewares.

### PR DESCRIPTION
**Pull request by anu-001**
#### What does this PR do?
Build signup for users with token and verification and decode middlewares
#### Description of Task to be completed?
Register a user
#### How should this be manually tested?
Clone repo and run npm install 

#### What are the relevant pivotal tracker stories?
[[ #157828773 ]](https://www.pivotaltracker.com/story/show/157828787)

### CODE REVIEW

**Review by ibrahim013 — COMMENTED** [review]
@Anuoluwa  why all the commented out codes?

File: server/app.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 import index from './db';
-import user from './routes/user';
+import auth from './routes/auth';
 import request from './routes/request';
 
+require('dotenv').config();
 
 const app = express();
-const port = 9000;
+const port = 3000;
```
Comment (ibrahim013) [inline server/app.js:11]: this will not work when hosting the app on Heroku

File: server/app.js [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 
+app.use('/auth', auth);
 app.use('/', request);
 // app.use('/users', user);
-
-
-app.listen(port, () => {
-  console.log(`app started at localost ${port}`);
-});
+if (!module.parent) { app.listen(port); }
```
Comment (ibrahim013) [inline server/app.js:18]: what is the` module.parent` for?

File: server/app.js (hunk 1 above)
Comment (anu-001) [reply server/app.js:11]: Ok, I will look into it

File: server/app.js (hunk 2 above)
Comment (anu-001) [reply server/app.js:18]: It due to an error, I had with my node, "Error: EARNINUSE",  I have removed it.

File: server/app.js (hunk 1 above)
Comment (anu-001) [reply server/app.js:11]: will it be better if I save it in  .env file? then I can use process.env.PORT where necessary.

**Pull request merged**